src/coreclr/tools/Common/TypeSystem/IL/Stubs/AsyncResumptionStub.cs

```diff
@@ -42,13 +42,201 @@ private MethodSignature InitializeSignature()
 
         public override MethodIL EmitIL()
         {
-            var emitter = new ILEmitter();
-            ILCodeStream codeStream = emitter.NewCodeStream();
+            ILEmitter ilEmitter = new ILEmitter();
+            ILCodeStream ilStream = ilEmitter.NewCodeStream();
 
-            // TODO: match getAsyncResumptionStub from CoreCLR VM
-            codeStream.EmitCallThrowHelper(emitter, Context.GetHelperEntryPoint("ThrowHelpers"u8, "ThrowNotSupportedException"u8));
+            // Ported from jitinterface.cpp CEEJitInfo::getAsyncResumptionStub
+            if (!_targetMethod.Signature.IsStatic)
+            {
+                if (_targetMethod.OwningType.IsValueType)
+                {
+                    ilStream.EmitLdc(0);
+                    ilStream.Emit(ILOpcode.conv_u);
+                }
+                else
+                {
+                    ilStream.Emit(ILOpcode.ldnull);
+                }
+            }
 
-            return emitter.Link(this);
+            if (Context.Target.Architecture != TargetArchitecture.X86)
+            {
+                if (_targetMethod.RequiresInstArg())
+                {
+                    ilStream.EmitLdc(0);
+                    if (Context.Target.PointerSize != 4)
+                        ilStream.Emit(ILOpcode.conv_i);
```

Comment:
```suggestion
                    ilStream.Emit(ILOpcode.conv_i);
```
There is no harm in emitting conv_i all the time.